iceoryx2-bb/threadsafe/src/trigger_queue.rs

```diff
@@ -47,19 +47,20 @@ pub use iceoryx2_bb_posix::semaphore::*;
 
 use core::{fmt::Debug, marker::PhantomData, time::Duration};
 use iceoryx2_bb_container::queue::FixedSizeQueue;
+use iceoryx2_bb_elementary::zero_copy_send::ZeroCopySend;
 use iceoryx2_bb_log::fatal_panic;
 
 const INTER_PROCESS_SUPPORT: bool = true;
 
 #[derive(Debug)]
-pub struct TriggerQueue<'a, T: Debug, const CAPACITY: usize> {
+pub struct TriggerQueue<'a, T: Debug + ZeroCopySend, const CAPACITY: usize> {
```

Comment:
Please do not restrict `T: ZeroCopySend` but implement `ZeroCopySend` manually without using the derive macro when `T` implements `ZeroCopySend`